float32 locX
float32 locY
float32 locZ